controllers/index.js

```diff
@@ -48,6 +48,26 @@ exports.index = (req, res) => {
         }
     }
 
+    // in a reverse proxy situation we have to use the referer to retrieve
+    // the correct protocol, hostname, and path
+    // unfortunately this won't work, when accessng the page directly:
+    // e.g.: http://example.com/hacktoberfest/?username=XXX
+    // in such a case we set hostname to an empty string and create the link
+    // with js after the page has loaded
+    if (req.headers['x-forwarded-for']) {
+        const referer = req.headers.referer;
+        if (referer) {
+            var hostname = referer.split('?')[0].slice(0, -1);
+            if (hostname.endsWith(statsLink.slice(0, -1))) {
```

Comment:
Can we rename hostname here, so `url` or something? Then we can leave the hostname definition above which you removed and we can lose an else.

Reply:
Yep, I see what you mean. Will do in a bit. Gonna grab lumch first. ;-)